### DISCUSSION TITLE
[Help]: Simple Sveltekit + Typescript question

### DISCUSSION THREAD

**Discussion by theRolento** [davidjerleke/embla-carousel#944] [Help] [answered]
### Summary

![immagine](https://github.com/user-attachments/assets/6adea097-79e2-4af4-a613-16a64b6bd6b4)
Which type is "event"? Nothing is wrote in the instructions here [https://www.embla-carousel.com/api/events/](https://www.embla-carousel.com/api/events/), but it's required

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [X] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [X] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment, marked as answer]
Hi @theRolento and thanks for your question.

The type can be found [here](https://github.com/davidjerleke/embla-carousel/blob/910a8dcb17e5b76521ec07c46d22347bfb123cca/packages/embla-carousel-svelte/src/components/emblaCarouselSvelte.ts#L19), so this should do it:

```html
<script>
  import emblaCarouselSvelte from 'embla-carousel-svelte'
  import { EmblaCarouselType } from 'embla-carousel'

  let emblaApi: EmblaCarouselType

  function onInit(event: CustomEvent<EmblaCarouselType>) {
    emblaApi = event.detail
    console.log(emblaApi.slideNodes()) // Access API
  }
</script>

<div
  class="embla"
  use:emblaCarouselSvelte
  on:emblaInit="{onInit}"
>
  <div class="embla__container">
    <div class="embla__slide">Slide 1</div>
    <div class="embla__slide">Slide 2</div>
    <div class="embla__slide">Slide 3</div>
  </div>
</div>
```

Let me know if it’s what you’re looking for. As you mentioned the type is missing [here in the docs](https://www.embla-carousel.com/api/events/#typescript). Feel free to **submit a PR** that adds the missing type. 

Best,
David

**Reply by theRolento** [reply]
Thank you!
